extension/src/experiments/model/index.ts

```diff
@@ -449,24 +450,24 @@ export class ExperimentsModel extends ModelWithPersistence {
     return this.numberOfCommitsToShow
   }
 
-  public setBranchesToShow(branches: string[]) {
-    this.branchesToShow = branches
+  public setBranches(currentBranch: string, allBranches: string[]) {
```

Comment:
I'm still seeing duplicate branches when I'm running an experiment:

https://github.com/iterative/vscode-dvc/assets/43496356/2aa590d8-b436-423c-9d34-bc3520ad05b7

Reply:
The top branch is on a different commit, what does the table look like before the experiment and after the experiment finishes?

Reply:
And what is the one branch that is selected?

Reply:
Oh, right I forgot that a detached HEAD is a different commit instead of a different branch. But I'm also seeing duplicates when I'm checked out to a different branch. Or maybe I'm misunderstanding the problem we're trying to solve 🤔 

I'm in a branch called `other-branch` and I have `main` selected. Before the experiment:

<img width="1297" alt="image" src="https://github.com/iterative/vscode-dvc/assets/43496356/10000a1b-ba35-4de1-9990-6adc9c22db48">

After the experiment:

<img width="1284" alt="image" src="https://github.com/iterative/vscode-dvc/assets/43496356/0b994d8c-c97c-41d5-bdf8-da5c179fcbfa">

Reply:
The problem that we are trying to solve here is that previously `other-branch` would be shown twice. This change shouldn't affect the other branches that are shown. It should only stop the detached head branch from being duplicated. does that make sense?

Reply:
Sorry for the description not being clearer.

Reply:
Oh, ok! Thanks for the clarification!